drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -457,11 +500,12 @@ def _is_finished(self):
         completed_count = 0
         total_count = 0
         item_state = self._get_item_state()
+
         for item in self.data['items']:
-            if item['zone'] != 'none':
+            item_id = item['id']
+            if self._get_valid_item_zones(item_id) != []:
                 total_count += 1
-                item_id = str(item['id'])
-                if item_id in item_state:
+                if str(item_id) in item_state:
                     completed_count += 1
```

Comment:
So I guess the question is, "what will the behavior of submitted, but incorrect, items be in Assessment Mode?" If they're added to `item_state`, but tagged as incorrect, then the logic for `_get_grade` would have to adapt to account for the fact that not all items in `item_state` are correct when running in assessment mode.

If we're fine with leaving incorrect answers out of `item_state`, then yeah, refactoring the logic  into another method makes sense.